app/src/main/java/org/freedombox/freedombox/views/adapter/DiscoveryListAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package org.freedombox.freedombox.views.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+import org.freedombox.freedombox.R
+
+class DiscoveryListAdapter(val context: Context, boxNameList: ArrayList<String>, portList: ArrayList<String>) : BaseAdapter() {
+
+    private var boxNameList: ArrayList<String>
+    private var portList: ArrayList<String>
+
+    init {
+        this.boxNameList = boxNameList
+        this.portList = portList
+    }
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
+        var inflaterView = convertView
```

Comment:
Use `val`

Reply:
Since `inflaterView` is a recycler view and hence cannot be set to `val`. Recycler view in the sense, if the view already exists it reuses or else uses the existing one.